pkg/kv/dynamodb/store.go

```diff
@@ -74,38 +73,41 @@ func (d *Driver) Open(ctx context.Context, kvParams kvparams.Config) (kv.Store,
 	if params == nil {
 		return nil, fmt.Errorf("missing %s settings: %w", DriverName, kv.ErrDriverConfiguration)
 	}
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+		Profile:           params.AwsProfile,
+	})
+	if err != nil {
+		return nil, err
+	}
 
-	var opts []func(*config.LoadOptions) error
-	if params.AwsRegion != "" {
-		opts = append(opts, config.WithRegion(params.AwsRegion))
+	cfg := aws.NewConfig()
+	if params.Endpoint != "" {
+		cfg.Endpoint = aws.String(params.Endpoint)
 	}
-	if params.AwsProfile != "" {
-		opts = append(opts, config.WithSharedConfigProfile(params.AwsProfile))
+
+	if params.AwsRegion != "" {
+		cfg = cfg.WithRegion(params.AwsRegion)
 	}
+
 	if params.AwsAccessKeyID != "" {
-		opts = append(opts, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(
-			params.AwsAccessKeyID,
-			params.AwsSecretAccessKey,
-			"",
-		)))
+		cfg = cfg.WithCredentials(credentials.NewCredentials(
+			&credentials.StaticProvider{
+				Value: credentials.Value{
+					AccessKeyID:     params.AwsAccessKeyID,
+					SecretAccessKey: params.AwsSecretAccessKey,
+				},
+			}))
 	}
 	const maxConnectionPerHost = 10
```

Comment:
We are bound to want to configure this to another value some time.

Reply:
True - I kept the current state in the current and previous release.